### PR TITLE
Side nav link text typing

### LINKED ISSUE #13605: Add TypeScript types to SideNavLinkText

### PR DESCRIPTION
**Pull request by OddDev**
Closes #13605 

Implementation of TypeScript definitions for components exported from @carbon/react. As discussed in #12513 

#### Changelog

**Changed**

- Type definitions of SideNavLinkText via TypeScript interface

#### Testing / Reviewing

Add a component of SideNavLinkText (or SideNavLink, respectively) and check whether all arguments/properties can be passed as expected.

### CODE REVIEW

**Comment by OddDev** [comment]
I have read the DCO document and I hereby sign the DCO.

File: packages/react/src/components/UIShell/SideNavLinkText.tsx [hunk 1]
```diff
@@ -20,16 +28,4 @@ function SideNavLinkText({ className: customClassName, children, ...rest }) {
...
-   * Provide the content for the link text
-   */
-  children: PropTypes.node.isRequired,
-
-  /**
-   * Provide an optional class to be applied to the containing node
-   */
-  className: PropTypes.string,
-};
-
```
Comment (guidari) [inline packages/react/src/components/UIShell/SideNavLinkText.tsx]: Can you just put it back the propTypes please. We still need that even with typescript. 
Other than that LGTM.

**Review by guidari — CHANGES_REQUESTED** [review]
Hey @OddDev Thanks for the contribution!

File: packages/react/src/components/UIShell/SideNavLinkText.tsx (hunk 1 above)
Comment (OddDev) [reply packages/react/src/components/UIShell/SideNavLinkText.tsx]: Will do :) Can you elaborate on why tho? (out of curiosity)

Comment (OddDev) [reply packages/react/src/components/UIShell/SideNavLinkText.tsx]: Done: 017ebd6

Comment (guidari) [reply packages/react/src/components/UIShell/SideNavLinkText.tsx]: It's useful for folks using JavaScript.
Here it is a few docs from React and from Carbon for more context:
https://react.dev/reference/react/Component#static-proptypes
https://github.com/carbon-design-system/carbon/blob/main/docs/guides/adding-component-types.md#strategy

**Review by guidari — APPROVED** [review]

**Review by alisonjoseph — APPROVED** [review]

**Pull request merged**